spec/features/app_spec.rb

```diff
@@ -513,3 +513,64 @@
     end
   end
 end
+
+# we test the metadata APIs for amazon
+describe 'metadata APIs' do
```

Comment:
Since there was an implementation question about the JSON responses, please add some tests on the new endpoints explicitly testing both XML and JSON formats, to prove the results are equivalent. Thanks.

Reply:
After that, it looks good to me.